### PR TITLE
chore(docs): Migrate link formats

### PR DESCRIPTION
**Pull request by ashleemboyer**
## What

- Migrates the formats of links in Markdown content as part of the [Docs Content Link Rewrites project](https://docs.google.com/document/d/1WaSyvoVPS-YCNiSPX0ynGpvc1gySEcpbrRoQVhimQCA/edit)
- Adds two new GitHub workflow files:
  - `.github/workflows/check-legacy-links-format.yml` - This is a workflow that checks if links in markdown content are in the new format. It does this by running the link rewrite script from `dev-portal`, and then asserting that no "rewriteable" links were found. This workflow will be used on subsequent PRs to maintain the new link format.
  - `.github/workflows/test-link-rewrites.yml` - This is a workflow that executes e2e tests of this PR's changes against the `main` branch's deploy preview. The tests only run for branches named `docs/amb.migrate-link-formats`, so future PRs will not have the tests run against them. This is a temporary workflow.

## Reviewing

With the two new workflows in place, this PR should only need a spot-check.

## Testing

- [ ] The latest run of the "Legacy Link Format Checker" workflow for this branch should pass ([check here](https://github.com/hashicorp/terraform-cdk/actions/workflows/check-legacy-links-format.yml))
- [ ] The latest run of the "Test Link Rewrites" workflow for this branch should pass ([check here](https://github.com/hashicorp/terraform-cdk/actions/workflows/test-link-rewrites.yml)), excluding:
  -  There is 1 test failing for the `/terraform/cdktf/api-reference/java` page. This is happening because the page is very large and failed to load in the test run. We know this because the `contentHrefs` being compared in the test are opposite: one is totally empty and the other has more than 10,000 links. Since this page's MDX file is not changed by this PR, we can ignore this failing test.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204081163003237

### CODE REVIEW

File: website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx
Comment (ansgarm) [inline website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx:9]: This link seems to be broken (and already was 😄) – I don't know what the link checker is going to catch, but this is definitely something our team should fix (either in this PR or in a separate one)

File: website/docs/cdktf/create-and-deploy/terraform-cloud.mdx
Comment (ashleemboyer) [inline website/docs/cdktf/create-and-deploy/terraform-cloud.mdx:183]: 🔴 Another off one: `https://terraform.io/concepts/remote-backends`

File: website/docs/cdktf/develop-custom-constructs/construct-design.mdx
Comment (ashleemboyer) [inline website/docs/cdktf/develop-custom-constructs/construct-design.mdx:29]: 🔴 Another! `https://terraform.io/develop-custom-constructs/publish-and-distribute#publish-to-public-or-private-registries`

File: website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx
Comment (ashleemboyer) [reply website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx:9]: I found 2 more! 😆 

My preference here would be for these links to be fixed outside of this PR. Otherwise, we'll see the tests for the files with broken links fail, because they'll be incorrect on the upstream deploy preview being tested as the source of truth.

If you'd like I can make that PR after we've got this one reviewed/approved/merged? 😊

**Comment by DanielMSchmidt** [comment]
We also need to adjust the links in our link shortener right?

**Comment by ashleemboyer** [comment]
> We also need to adjust the links in our link shortener right?

@DanielMSchmidt I'm not sure what you're referring to! Might be something I haven't seen yet 😊

File: website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx
Comment (ansgarm) [reply website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx:9]: Ah, got it! Sure, no problem! A PR would be the best, but a list of broken links is fine, too.

Comment (ansgarm) [reply website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx:9]: just saw your other comments 😄

**Review by ansgarm — APPROVED** [review]
Looks good to be merged by the way 🎉
(besides the three broken links that are already broken currently and which we will fix after this 👍)

**Comment by ashleemboyer** [comment]
> We link to some of these pages here: https://github.com/hashicorp/terraform-cdk/blob/main/cdk.tf/vercel.json
> 
> (edit: They all redirect from cdk.tf to various destinations)

@ansgarm ah okay, I see! @BRKalow or @thiskevinwang can probably provide stronger feedback on this since they have more familiarity with it.

**Comment by ansgarm** [comment]
That is something that is managed by us, I think updating the links that point to developer.hashicorp.com would be enough, but could also be done in a follow-up PR 😇

**Comment by ashleemboyer** [comment]
@ansgarm my account doesn't have the permissions to merge this PR, so feel free to do it when you're ready!

**Comment by ansgarm** [comment]
@ashleemboyer 👋 Happy to merge this, however the "Test Link Rewrites" is still failing (but as I read in Slack, that is okay?) – it also appears that this workflow is only intended to run on this PR to check this change.
Given that we don't really need it. Can we maybe remove it as part of this PR already? Saves us the extra PR 😄

**Comment by ansgarm** [comment]
Mergeddddd 💯

I kept the discussion items open for a follow-up PR to fix the broken links

File: website/docs/cdktf/develop-custom-constructs/construct-design.mdx
Comment (ansgarm) [reply website/docs/cdktf/develop-custom-constructs/construct-design.mdx:29]: fixed here: https://github.com/hashicorp/terraform-cdk/pull/2693

File: website/docs/cdktf/create-and-deploy/terraform-cloud.mdx
Comment (ansgarm) [reply website/docs/cdktf/create-and-deploy/terraform-cloud.mdx:183]: fixed here: https://github.com/hashicorp/terraform-cdk/pull/2693

File: website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx
Comment (ansgarm) [reply website/docs/cdktf/develop-custom-constructs/publishing-and-distribution.mdx:9]: fixed here: https://github.com/hashicorp/terraform-cdk/pull/2693

**Pull request merged**
